plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiConfig.java

```diff
@@ -0,0 +1,249 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.hudi;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.DataSize;
+
+import javax.validation.constraints.DecimalMax;
+import javax.validation.constraints.DecimalMin;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import java.util.List;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.util.Locale.ENGLISH;
+import static org.apache.hudi.common.model.HoodieFileFormat.PARQUET;
+
+public class HudiConfig
+{
+    private static final Splitter COMMA_SPLITTER = Splitter.on(",").omitEmptyStrings().trimResults();
+
+    private String baseFileFormat = PARQUET.name();
+    private List<String> columnsToHide = ImmutableList.of();
+    private boolean metadataEnabled;
+    private boolean shouldSkipMetaStoreForPartition;
+    private boolean shouldUseParquetColumnNames = true;
+    private int partitionScannerParallelism = 16;
+    private int splitGeneratorParallelism = 16;
+    private int minPartitionBatchSize = 10;
+    private int maxPartitionBatchSize = 100;
+    private boolean sizeBasedSplitWeightsEnabled = true;
+    private DataSize standardSplitWeightSize = DataSize.of(128, MEGABYTE);
+    private double minimumAssignedSplitWeight = 0.05;
+    private int maxSplitsPerSecond = Integer.MAX_VALUE;
+    private int maxOutstandingSplits = 1000;
+
+    @NotNull
+    public String getBaseFileFormat()
+    {
+        return baseFileFormat;
+    }
+
+    @Config("hudi.base-file-format")
+    public HudiConfig setBaseFileFormat(String baseFileFormat)
+    {
+        this.baseFileFormat = baseFileFormat;
+        return this;
+    }
+
+    public List<String> getColumnsToHide()
+    {
+        return columnsToHide;
+    }
+
+    @Config("hudi.columns-to-hide")
+    public HudiConfig setColumnsToHide(String columnsToHide)
+    {
+        this.columnsToHide = COMMA_SPLITTER.splitToStream(nullToEmpty(columnsToHide))
+                .map(s -> s.toLowerCase(ENGLISH))
+                .collect(toImmutableList());
+        return this;
+    }
+
+    @Config("hudi.metadata-enabled")
+    @ConfigDescription("Fetch the list of file names and sizes from metadata rather than storage")
+    public HudiConfig setMetadataEnabled(boolean metadataEnabled)
+    {
+        this.metadataEnabled = metadataEnabled;
+        return this;
+    }
+
+    @NotNull
+    public boolean isMetadataEnabled()
+    {
+        return this.metadataEnabled;
+    }
+
+    @Config("hudi.skip-metastore-for-partition")
+    @ConfigDescription("By default, partition info is fetched from the metastore. " +
```

Comment:
How would this custom extractor be used in Trino? We don't want users to have to add custom code.

This config property needs to be covered by integration tests, so that we test both code paths. If it's too hard to test, then let's remove it for now.